app/routes.py

```diff
@@ -0,0 +1,218 @@
+		if user is None or not user.check_password(password):
+			flash("User doesnt exist or password is incorrect")
+			return redirect(url_for('login_get'))
+		else:
+			login_user(user)								# Authentication of User
+			session['user_id'] = user.id
+			return redirect('userhome')
+		
+
+# User Home Page
+@app.route('/userhome')
+@login_required
+def userhome():
+	# Get the tasks to homepage
+	# Retreive data
+	tasks = current_user.tasks.all()
+	print(tasks)
+	
+	if len(tasks) > 0 :
+		# return render_template('userhome.html', tasks = tasks)
+		response = make_response(render_template('userhome.html', tasks = tasks))
+		response.headers['Last-Modified'] = datetime.now()
+		response.headers['Cache-Control'] = 'no-store, no-cache, must-revalidate, post-check=0, pre-check=0, max-age=0'
+		response.headers['Pragma'] = 'no-cache'
+		response.headers['Expires'] = '-1'
+		return response
+	else:
+		response = make_response(render_template('userhome.html'))
+		response.headers['Last-Modified'] = datetime.now()
+		response.headers['Cache-Control'] = 'no-store, no-cache, must-revalidate, post-check=0, pre-check=0, max-age=0'
+		response.headers['Pragma'] = 'no-cache'
+		response.headers['Expires'] = '-1'
+		return response
+	# Close cursor
+	cur.close()
+
+# Add Task
+@app.route('/addTask', methods = ['GET'])
+@login_required
+def addTask_get():
+	form = AddTaskForm()
+	return render_template('addTask.html', form=form)
+
+@app.route('/addTask', methods = ['POST'])
+@login_required
+def addTask_post():
+	form = AddTaskForm(request.form)
+	if form.validate():
+		form = AddTaskForm(request.form)	
+		# Get Task Details
+		title = form.title.data
+		body = form.body.data
+		status = False
+
+		# Insert tasks into database
+		task = TaskList(title=title, body=body, status=status, user_id=current_user.id)
+		db.session.add(task)
+		db.session.commit()
+		
+		flash("Task added successfully", "success")
+		return redirect(url_for('userhome'))
+
+	return render_template('addTask.html', form=form)
+
+# Edit task
+@app.route('/editTask/<string:id>', methods = ['GET'])
+@login_required
+def editTask_get(id):
+	# Get data
+	task = TaskList.query.filter_by(id=id).first()
+	
+	if task is not None and task.is_own_task(task.user_id):
+		return render_template('editTask.html', task=task)
+	else :
+		flash("The task doesnot belong to you", "danger")
+		return redirect(url_for('userhome'))
+
+@app.route('/editTask/<string:id>', methods = ['POST'])
+@login_required
+def editTask_post(id):
+	# Get data
+	task = TaskList.query.filter_by(id=id).first()
+	
+	# Get Task Details
+	task.title = request.form['title']
+	task.body = request.form['body']
+	task.status = False
+
+	db.session.add(task)
+	db.session.commit()
+
+	flash("Task Updated successfully", "success")
+
+	return redirect(url_for('userhome'))
+
+# Delete Task
+@app.route('/deleteTask/<string:id>', methods=['POST'])
+@login_required
+def deleteTask(id):
+	task = TaskList.query.filter_by(id=id).first()
```

Comment:
Add `user_id` owner check here. Any logged in user would be able to delete any task.

Reply:
Not necessary. Another user canot go to that page by typing the link
It automatically redirects.